### PR TITLE
Adjusted AI calcs for Triple Kick Effect

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Updates AI calcs for the Triple Kick Effect to match it closer to actual damage

### CODE REVIEW

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -605,11 +605,25 @@ struct SimulatedDamage AI_CalcDamage(u32 move, u32 battlerAtk, u32 battlerDef, u
...
         else
         {
-            s32 nonCritDmg = CalculateMoveDamageVars(move, battlerAtk, battlerDef, moveType, fixedBasePower,
-                                                    effectivenessMultiplier, weather, FALSE,
-                                                    aiData->holdEffects[battlerAtk], aiData->holdEffects[battlerDef],
-                                                    aiData->abilities[battlerAtk], aiData->abilities[battlerDef]);
+            s32 nonCritDmg = 0;
+            if (gMovesInfo[move].effect == EFFECT_TRIPLE_KICK)
+            {
+                for (gMultiHitCounter = 3; gMultiHitCounter > 0; gMultiHitCounter--)
```
Comment (Bassoonian) [inline src/battle_ai_util.c:611]: I may be missing something trivial, but is there a reason we're using a global here instead of a local variable?

Comment (AlexOn1ine) [reply src/battle_ai_util.c:611]: yes, the global is needed for this part
```
case EFFECT_TRIPLE_KICK:
    basePower *= 1 + gMovesInfo[move].strikeCount - gMultiHitCounter;
    break;
```
Usually what happens is the at the start of the move execution `gMultiHitCounter` is set for multi hit moves and decremented for each hit. There was previously a had that just multiplied the triple kick damage by 6 to approximate it but it was not accurate.

What this change does is simulate it with `gMultiHitCounter`

File: src/battle_ai_util.c [hunk 2]
```diff
@@ -605,11 +605,25 @@ struct SimulatedDamage AI_CalcDamage(u32 move, u32 battlerAtk, u32 battlerDef, u
...
+                {
+                    nonCritDmg += CalculateMoveDamageVars(move, battlerAtk, battlerDef, moveType, fixedBasePower,
+                                                            effectivenessMultiplier, weather, FALSE,
+                                                            aiData->holdEffects[battlerAtk], aiData->holdEffects[battlerDef],
+                                                            aiData->abilities[battlerAtk], aiData->abilities[battlerDef]);
```
Comment (Bassoonian) [inline src/battle_ai_util.c:616]: indentation

File: src/battle_ai_util.c [hunk 3]
```diff
@@ -605,11 +605,25 @@ struct SimulatedDamage AI_CalcDamage(u32 move, u32 battlerAtk, u32 battlerDef, u
...
+                }
+            }
+            else
+            {
+                nonCritDmg = CalculateMoveDamageVars(move, battlerAtk, battlerDef, moveType, fixedBasePower,
+                                                        effectivenessMultiplier, weather, FALSE,
+                                                        aiData->holdEffects[battlerAtk], aiData->holdEffects[battlerDef],
+                                                        aiData->abilities[battlerAtk], aiData->abilities[battlerDef]);
```
Comment (Bassoonian) [inline src/battle_ai_util.c:624]: indentation

File: src/battle_ai_util.c (hunk 1 above)
Comment (Bassoonian) [reply src/battle_ai_util.c:611]: Shouldn't `gMultiHitCounter = 3` be `gMultiHitCounter = gMovesInfo[move].strikeCount` then?

Comment (AlexOn1ine) [reply src/battle_ai_util.c:611]: I can add a comment just in case somebody else is surprised

**Comment by AlexOn1ine** [comment]
> Shouldn't gMultiHitCounter = 3 be gMultiHitCounter = gMovesInfo[move].strikeCount then?

It doesn't matter in this case but ye that would make more sense.

**Comment by Bassoonian** [comment]
> > Shouldn't gMultiHitCounter = 3 be gMultiHitCounter = gMovesInfo[move].strikeCount then?
> 
> It doesn't matter in this case but ye that would make more sense.

If we're going to avoid using the hardcoded `3` in some places we may as well be consistent :)

**Comment by AlexOn1ine** [comment]
Ready for review. 

I noticed that there is a lot of duplicate code in this function which might be worth to clean up but AI calcs already cause a delay so I'm not sure if introducing a new function will make it worse. I'll have to talk with Egg about this.

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**